module/os/linux/zfs/arc_os.c

```diff
@@ -80,10 +82,20 @@ static struct notifier_block arc_hotplug_callback_mem_nb;
 
 /*
  * Return a default max arc size based on the amount of physical memory.
+ * This may be overridden by tuning the zfs_arc_max module parameter.
  */
 uint64_t
 arc_default_max(uint64_t min, uint64_t allmem)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0)
```

Comment:
Why should this be limited to 5.15 and newer kernels? What specifically changed?

Reply:
This can be changed as needed. There is nothing magic to 5.15 other than being in the recent Ubuntu.  Main idea is not to go w/older kernels since newer kernels could have already fixed/addressed an issue.  I think of it as a bit of a safety net, tho the line could be drawn elsewhere or removed entirely if that would be more appropriate.

Reply:
I would also like to follow this with improvements to page cache integration

Reply:
We should keep the default value the same across kernel versions unless we have a good reason to set it differently.  It sounds like we don't.  However, if we did we'd want to specify the relevant kernel version and explain why in the `zfs(4)` man page. 

> I would also like to follow this with improvements to page cache integration. Do you think they should be tied together?

I don't think they need to be tied together, but I suspect this change will break things for some users using libvirt with ZFS.  This is really the use case I'm most concerned with and we should test before making this change.  I suspect without some kind of page cache integration manually tuning the ARC back to a smaller value will be needed for this libvirt workload.

Reply:
Version constraint removed.